llama_index/program/llm_program.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Dict, Generic, Optional, Type, Union
+
+from llama_index.bridge.langchain import ChatOpenAI, HumanMessage
+
+from llama_index.types import Model
+from llama_index.program.base_program import BasePydanticProgram
+from llama_index.prompts.base import Prompt
+from llama_index.bridge.langchain import BaseLanguageModel
+from llama_index.output_parsers.pydantic import PydanticOutputParser
+
+
+SUPPORTED_MODEL_NAMES = [
+    "gpt-3.5-turbo-0613",
+    "gpt-4-0613",
+]
+
+
+class LLMTextCompletionProgram(BasePydanticProgram, Generic[Model]):
+    """
+    LLM Text Completion Program.
+
+    Uses generic LLM text completion + an output parser to generate a structured output.
+
+    """
+
+    def __init__(
+        self,
+        output_parser: PydanticOutputParser,
+        prompt: Prompt,
+        llm: BaseLanguageModel,
+        function_call: Union[str, Dict[str, Any]],
+        verbose: bool = False,
+    ) -> None:
+        self._output_parser = output_parser
+        self._llm = llm
+        self._prompt = prompt
+        self._verbose = verbose
+        self._function_call = function_call
+
+    @classmethod
+    def from_defaults(
+        cls,
+        output_parser: PydanticOutputParser,
+        prompt_template_str: str,
+        llm: Optional[BaseLanguageModel] = None,
+        verbose: bool = False,
+        function_call: Optional[Union[str, Dict[str, Any]]] = None,
+        **kwargs: Any,
+    ) -> "LLMTextCompletionProgram":
+        llm = llm or ChatOpenAI(temperature=0, model_name="gpt-3.5-turbo-0613")
+        if not isinstance(llm, ChatOpenAI):
+            raise ValueError("llm must be a ChatOpenAI instance")
+
+        if llm.model_name not in SUPPORTED_MODEL_NAMES:
+            raise ValueError(
+                f"Model name {llm.model_name} not supported. "
+                f"Supported model names: {SUPPORTED_MODEL_NAMES}"
```

Comment:
should remove this

Reply:
I agree, it's not using anything specific to these models

Reply:
yeah you're right i was just copy/pasting code :)